File: documentation/user/en/query/requirements/examples/facet/facet-groups-disjunction.cs
EvitaResponse<ISealedEntity> entities = evita.QueryCatalog(
	"evita",
	session => session.QuerySealedEntity(
        Query(
        	Collection("Product"),
        	FilterBy(
        		UserFilter(
        			FacetHaving(
        				"parameterValues",
        				EntityHaving(
        					AttributeInSet("code", "ram-memory-64")
        				)
        			)
        		)
        	),
        	Require(
        		FacetSummaryOfReference(
        			"parameterValues",
        			Impact,
        			FilterBy(
        				AttributeContains("code", "4")
        			),
        			FilterGroupBy(
        				AttributeInSet("code", "ram-memory", "rom-memory")
        			),
        			EntityFetch(
        				AttributeContent("code")
        			),
        			EntityGroupFetch(
        				AttributeContent("code")
        			)
        		),
        		FacetGroupsDisjunction(
        			"parameterValues",
        			FilterBy(
        				AttributeInSet("code", "ram-memory", "rom-memory")
        			)
        		)
        	)
        )
	)
);

File: documentation/user/en/query/requirements/examples/facet/facet-groups-disjunction.java
final EvitaResponse<SealedEntity> entities = evita.queryCatalog(
	"evita",
	session -> {
		return session.querySealedEntity(
			query(
				collection("Product"),
				filterBy(
					userFilter(
						facetHaving(
							"parameterValues",
							entityHaving(
								attributeInSet("code", "ram-memory-64")
							)
						)
					)
				),
				require(
					facetSummaryOfReference(
						"parameterValues",
						IMPACT,
						filterBy(
							attributeContains("code", "4")
						),
						filterGroupBy(
							attributeInSet("code", "ram-memory", "rom-memory")
						),
						entityFetch(
							attributeContent("code")
						),
						entityGroupFetch(
							attributeContent("code")
						)
					),
					facetGroupsDisjunction(
						"parameterValues",
						WITH_DIFFERENT_GROUPS,
						filterBy(
							attributeInSet("code", "ram-memory", "rom-memory")
						)
					)
				)
			)
		);
	}
);
